Assets/Scripts/WaveManager.cs

```diff
@@ -136,13 +140,14 @@ void Spawn(Attacker attacker, Vector3 center)
         Vector3 surrounding = new Vector3(Mathf.Sin(angle), Mathf.Cos(angle), 0);
         surrounding *= range;
         surrounding += center;
-
-        //Instantiate prefab, register attacker and let it target the player.
-        GameObject parent = attacker.transform.gameObject;
-        parent.GetComponent<AIDestinationSetter>().target = player.transform;
-        attacker.target = player.GetComponent<Damagable>();
-
-        GameObject spawned = Instantiate(parent, surrounding, Quaternion.identity);
+        
+        GameObject spawned = Instantiate(attacker.transform.gameObject, surrounding, Quaternion.identity);
+        
+        // Give Players to Aggro finder of instance;
+        var aggro = spawned.GetComponent<EnemyAggro>();

```

Comment:
Please declare the type as it is done in the rest of the file.
And could you move the code of the if statement into a separate line.
It honestly is hard to read that way.